tree/treeplayer/inc/TTreeReaderArray.h

```diff
@@ -59,90 +59,159 @@ namespace Internal {
 } // namespace ROOT
 
 template <typename T>
-class TTreeReaderArray: public ROOT::Internal::TTreeReaderArrayBase {
+class TTreeReaderArray : public ROOT::Internal::TTreeReaderArrayBase {
 public:
+   /// Create an array reader of branch "branchname" for TTreeReader "tr".
+   TTreeReaderArray(TTreeReader &tr, const char *branchname)
+      : TTreeReaderArrayBase(&tr, branchname, TDictionary::GetDictionary(typeid(T))) {}
+
+   T &At(std::size_t idx) { return *static_cast<T *>(UntypedAt(idx)); }
+   const T &At(std::size_t idx) const { return *static_cast<T *>(UntypedAt(idx)); }
+   T &operator[](std::size_t idx) { return At(idx); }
+   const T &operator[](std::size_t idx) const { return At(idx); }
+
+   /// Random access iterator to the elements of a TTreeReaderArray
+   class Iterator_t : public std::iterator<std::random_access_iterator_tag, T> {
```

Comment:
I prefer to keep the order: types, data members, function members. That's because fairly often the types are needed by the members, and having a common order helps readability (the long version of "our (future) coding rules say so).

Reply:
changed the order of TTreeReaderArray internals to Iterator_t definition, using declarations, methods.